node/src/command.rs

```diff
@@ -30,76 +35,149 @@ pub fn run() -> sc_cli::Result<()> {
     match &cli.subcommand {
         Some(Subcommand::Benchmark(cmd)) => {
             let runner = cli.create_runner(cmd)?;
+            let chain_spec = &runner.config().chain_spec;
 
-            runner.sync_run(|config| {
-                let PartialComponents { client, backend, .. } =
-                    new_partial::<RuntimeApi, ExecutorDispatch>(&config)?;
-
+            match cmd {
                 // This switch needs to be in the client, since the client decides
                 // which sub-commands it wants to support.
-                match cmd {
-                    BenchmarkCmd::Pallet(cmd) => {
-                        if !cfg!(feature = "runtime-benchmarks") {
-                            return Err("Runtime benchmarking wasn't enabled when building the \
-                                        node. You can enable it with `--features \
-                                        runtime-benchmarks`."
-                                .into());
-                        }
-
-                        cmd.run::<Block, ExecutorDispatch>(config)
+                BenchmarkCmd::Pallet(cmd) => {
+                    if !cfg!(feature = "runtime-benchmarks") {
+                        return Err("Runtime benchmarking wasn't enabled when building the node. \
+                                    You can enable it with `--features runtime-benchmarks`."
+                            .into());
                     }
-                    BenchmarkCmd::Block(cmd) => cmd.run(client),
-                    BenchmarkCmd::Storage(cmd) => {
-                        let db = backend.expose_db();
-                        let storage = backend.expose_storage();
 
-                        cmd.run(config, client, db, storage)
+                    match chain_spec {
+                        #[cfg(feature = "with-zeitgeist-runtime")]
+                        spec if spec.is_zeitgeist() => runner.sync_run(|config| {
+                            cmd.run::<zeitgeist_runtime::Block, ZeitgeistExecutor>(config)
+                        }),
+                        #[cfg(feature = "with-battery-station-runtime")]
+                        _ => runner.sync_run(|config| {
+                            cmd.run::<battery_station_runtime::Block, BatteryStationExecutor>(
+                                config,
+                            )
+                        }),
+                        #[cfg(not(feature = "with-battery-station-runtime"))]
+                        _ => panic!("Invalid chain spec"),
```

Comment:
Suggestion: may be more detailed message with suggestion to use either feature , similar to line 45.

Reply:
Mitigated in [0509243](https://github.com/zeitgeistpm/zeitgeist/pull/721/commits/05092431ee6504012d89ff54862961e7bfafac5d)